llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -1366,6 +1383,67 @@ bool SPIRVInstructionSelector::selectRsqrt(Register ResVReg,
       .constrainAllUses(TII, TRI, RBI);
 }
 
+// Since there is no integer dot implementation, expand by piecewise multiplying
```

Comment:
actually I see a `OpSDot`, `OpUDot`, and `OpSUDot` (for mixes signedness)  in the [spec](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html). Maybe we just don't use these in DXC's spirv generation. I'm fine with sticking with what DXC does we should at least mark the comment accordingly.

Reply:
Those are fairly recent SPIRV extensions. I didn't think incorporating them was within scope.